### PR TITLE
Add is_inside func to ImageSpec

### PR DESCRIPTION
**Pull request by pingsutw**
# TL;DR

`is_inside` will check if current image is built from image_spec when running remotely. This function always returns true when running locally, which means it will import all the package in local execution.

```python
from flytekit import task, workflow, Resources
from flytekit.image_spec import ImageSpec

new_flytekit = "git+https://github.com/flyteorg/flytekit@db448b054533de52e3a2d7fdb08be31e7b8a5c7d"
tensorflow_image_spec = ImageSpec(packages=["tensorflow", new_flytekit], apt_packages=["git"], registry="pingsutw")
torch_image_spec = ImageSpec(packages=["torch", new_flytekit], apt_packages=["git"], registry="pingsutw")

if tensorflow_image_spec.in_container():
    print("import tensorflow")
    import tensorflow  # import tensorflow only when running t1

if torch_image_spec.in_container():
    print("import torch")
    import torch  # import torch only when running t2


@task(container_image=tensorflow_image_spec, requests=Resources(mem="900Mi"))
def t1():
    tensorflow.constant(4)
    print("tensorflow")


@task(container_image=torch_image_spec, requests=Resources(mem="900Mi"))
def t2():
    torch.tensor([[1., -1.], [1., -1.]])
    print("torch")


@workflow
def wf():
    t1()
    t2()


if __name__ == '__main__':
    wf()
```

```python
from flytekit import task, workflow, Resources
from flytekit.image_spec import ImageSpec

new_flytekit = "git+https://github.com/flyteorg/flytekit@6a1dfaea2e2521c9f9a009cf1e9cefdc97c27aed"
tensorflow_image_spec = ImageSpec(packages=["tensorflow", new_flytekit], apt_packages=["git"], registry="pingsutw")
torch_image_spec = ImageSpec(packages=["torch", new_flytekit], apt_packages=["git"], registry="pingsutw")

if tensorflow_image_spec.is_container():
    print("import tensorflow")
    import tensorflow


    @task(container_image=tensorflow_image_spec, requests=Resources(mem="900Mi"))
    def t1() -> tensorflow.constant:
        t = tensorflow.constant(4)
        print("tensorflow")
        return t

if torch_image_spec.is_container():
    print("import torch")
    import torch

    @task(container_image=torch_image_spec, requests=Resources(mem="900Mi"))
    def t2() -> torch.tensor:
        t = torch.tensor([[1., -1.], [1., -1.]])
        print("torch")
        return t


@workflow
def wf():
    t1()
    t2()


if __name__ == '__main__':
    wf()
```

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
^^^

## Tracking Issue
_NA_

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit/image_spec/image_spec.py [hunk 1]
```diff
@@ -52,6 +54,11 @@ def image_name(self) -> str:
             container_image = f"{self.registry}/{container_image}"
         return container_image
 
+    def is_inside(self) -> bool:
```
Comment (kumare3) [inline flytekit/image_spec/image_spec.py:57]: Shall we call it - in_container?

File: plugins/flytekit-envd/flytekitplugins/envd/image_builder.py [hunk 2]
```diff
@@ -39,7 +39,7 @@ def create_envd_config(image_spec: ImageSpec) -> str:
     packages = [] if image_spec.packages is None else image_spec.packages
     apt_packages = [] if image_spec.apt_packages is None else image_spec.apt_packages
     env = {} if image_spec.env is None else image_spec.env
-    env.update({"PYTHONPATH": "/root"})
+    env.update({"PYTHONPATH": "/root", FLYTE_IMAGE_NAME: image_spec.image_name()})
```
Comment (kumare3) [inline plugins/flytekit-envd/flytekitplugins/envd/image_builder.py:42]: Let's make the env car harder to overwrite by mistake 

_F_IMG_ID

File: flytekit/image_spec/image_spec.py (hunk 1 above)
Comment (wild-endeavor) [reply flytekit/image_spec/image_spec.py]: can we do
```
is = ImageSpec(...)
with is:
    import blah
```

File: flytekit/image_spec/image_spec.py [hunk 3]
```diff
@@ -52,6 +54,11 @@ def image_name(self) -> str:
             container_image = f"{self.registry}/{container_image}"
         return container_image
 
+    def is_container(self) -> bool:
+        if os.environ.get(_F_IMG_ID):
```
Comment (kumare3) [inline flytekit/image_spec/image_spec.py:58]: if will return true in all cases where `_F_IMG_ID` is not set
For example 
```python
@task 
def foo():
  ...

i = ImageSpec(...)

if i.in_container():
   @task(container_image=)
   def bar()
      ...
```

When task `foo` is loaded at runtime, then `bar` will also be loaded right?
should we also add 
```
if not os.environ.get(_F_IMG_ID):
  if current_context().execution_id:
       return False
  return True
else return os.environ.get(_F_IMG_ID) == self.image_name()
```

**Review by kumare3 — APPROVED** [review]

**Pull request merged**